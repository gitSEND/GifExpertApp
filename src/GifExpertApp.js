import React, { useState } from "react";
import { AddCategory } from "./compoents/AddCategory";
import { GifGrid } from "./compoents/GifGrid";

export const GifExpertApp = () => {

	const [categories, setCategories] = useState(["Dragon Ball"]);

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories}></AddCategory>
			<hr></hr>
			<ol>
				{
					categories.map((category) => (
						<GifGrid
							key={category}
							category={category}>
						</GifGrid>
					))
				}
			</ol>
		</>
	);
};
