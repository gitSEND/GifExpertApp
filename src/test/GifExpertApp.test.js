
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Probar componente <GifExpertApp></GifExpertApp>', () => {


    test('debe mostrarse correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp></GifExpertApp>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['naruto', 'goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })



})

