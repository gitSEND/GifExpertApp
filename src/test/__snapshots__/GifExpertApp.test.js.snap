// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar componente <GifExpertApp></GifExpertApp> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="naruto"
      key="naruto"
    />
    <GifGrid
      category="goku"
      key="goku"
    />
  </ol>
</Fragment>
`;

exports[`Probar componente <GifExpertApp></GifExpertApp> debe mostrarse correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
